weld/src/main/java/org/jboss/as/weld/WeldExtension.java

```diff
@@ -111,7 +112,7 @@ protected boolean rejectAttribute(PathAddress address, String attributeName, Mod
                         //This will not get called if it was discarded, so reject if it is undefined (default==false) or if defined and != 'true'
                         return !attributeValue.isDefined() || !attributeValue.asString().equals("true");
                     }
-                }, WeldResourceDefinition.NON_PORTABLE_MODE_ATTRIBUTE, WeldResourceDefinition.REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE)
+                }, WeldResourceDefinition.NON_PORTABLE_MODE_ATTRIBUTE, WeldResourceDefinition.REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE, WeldResourceDefinition.DEVELOPMENT_MODE_ATTRIBUTE)
```

Comment:
Yes, the attribute is assumed false unless explicitly defined in the new version. This is the same for all the attributes defined by the weld subsystem so far. I am not sure why the comment says: "These new attributes are assumed to be 'true' in the old version but default to false" - the "true in old version" part is wrong.